pkg/artifacts/fetch.go

```diff
@@ -148,41 +172,38 @@ func FetchLastTestResult(test_matrix v1.MatrixSpec, matrix_name string, test v1.
 	last_test_file, err := fetchTestResult(test_matrix, test.ProwName,
 		string(last_test_build_id), filename, filetype)
 	if (err != nil) {
-		return "", ArtifactResult{}, fmt.Errorf("error fetching the results of the last test of %s:%s (%s): %v",
+		return "", ArtifactResult{}, fmt.Errorf("error fetching the results of the last test of %s: %s (%s): %v",
 			matrix_name, test.ProwName, last_test_build_id, err)
 
 	}
 
 	return string(last_test_build_id), last_test_file, nil
 }
 
-
-func FetchLastNTestResults(test_matrix v1.MatrixSpec, matrix_name, prow_name string, nb_test int, filename string, filetype ArtifactType) ([]string, map[string]ArtifactResult, error) {
-	test_list_path := fmt.Sprintf("%s/?index", prow_name)
+func FetchLastNTestResults(test_matrix *v1.MatrixSpec, matrix_name, prow_name string, test_history int, filename string, filetype ArtifactType) ([]string, map[string]ArtifactResult, error) {
+	test_list_path := fmt.Sprintf("%s/", prow_name)
 	test_list_html, err := fetchHtmlArtifact(test_matrix, test_list_path)
 	if err != nil {
 		return nil, nil, fmt.Errorf("error fetching the tests of %s / %s: %v", matrix_name, prow_name, err)
 	}
 
 	test_results := map[string]ArtifactResult{}
-	build_ids := make([]string, 0, nb_test)
 
-	test_list_html.Find("li.grid-row").EachWithBreak(func(i int, s *goquery.Selection) bool {
-		entry_type, found := s.Find("img").Attr("src")
-		if !found || entry_type != "/icons/dir.png" {
-			return true
-		}
-		test_build_id := strings.TrimSuffix(strings.TrimSpace(s.Find("a").Text()), "/")
-		if (!found) {
-			return true
-		}
+	build_ids, err := listFilesInDirectory(test_list_html, true, false)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error fetching last test results: %v", err)
+	}
 
-		build_ids = append([]string{test_build_id}, build_ids...)
-		return true
-	})
-	if len(build_ids) > nb_test {
-		build_ids = build_ids[:nb_test]
+	// `build_ids` order is "oldest first" (alphanumeric order of timestamps)
+
+	if len(build_ids) > test_history {
+		build_ids = build_ids[len(build_ids) - test_history:]
 	}
+
+	build_ids = reverseStringArray(build_ids)
+
+	// `build_ids` order is now "newest first"
+
```

Comment:
Perhaps an extra line here (`\n`), but this is a matter of style/preference.  Similarly above.

Reply:
it's on purpose here, it's a kind of "assert statement",

> assert sorted(build_ids, newest_first) == build_ids